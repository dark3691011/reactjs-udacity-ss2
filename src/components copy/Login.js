import login_img from "../assets/login_img.jpg";

const Login = () => {
  return (
    <div className="d-flex flex-column justify-center align-items-center ">
      <h1>Employee Polls</h1>
      <img src={login_img} />
      <div className="d-flex flex-column justify-center align-items-center w-100">
        <h3>Login</h3>
        <div className="d-flex flex-column justify-center align-items-center w-100">
          <label className="p-3">Username</label>
          <input className="w-100 input-login" placeholder="Username" />
        </div>
        <div className="d-flex flex-column justify-center align-items-center w-100">
          <label className="p-3 ">Password</label>
          <input className="w-100 input-login" placeholder="Password" />
        </div>
        <div className="login-button ">Submit</div>
      </div>
    </div>
  );
};

export default Login;
