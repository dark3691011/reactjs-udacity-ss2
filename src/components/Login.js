import { connect } from "react-redux";
import { handleLogin } from "../actions/authedUser";
import { useNavigate } from "react-router-dom";
import { useState } from "react";
import login_img from "../assets/login_img.jpg";

const Login = ({ dispatch }) => {
  const navigate = useNavigate();
  const [userName, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState("");
  const handleUserNameChange = (e) => {
    const text = e.target.value;
    setUserName(text);
  };
  const handlePasswordChange = (e) => {
    const text = e.target.value;
    setPassword(text);
  };

  const login = (e) => {
    e.preventDefault();
    dispatch(
      handleLogin({
        userName,
        password,
      })
    ).then((res) => {
      if (res) {
        navigate("/");
      }
      setError(`Invalid username or password`); // Set the error message if login fails
    });
  };

  return (
    <div className="d-flex flex-column justify-center align-items-center ">
      <h1>Employee Polls</h1>
      <img src={login_img} />
      <div className="d-flex flex-column justify-center align-items-center w-100">
        <h3>Login</h3>
        {error && <p style={{ color: "red" }}>{error}</p>}{" "}
        {/* Display error message */}
        <div className="d-flex flex-column justify-center align-items-center w-100">
          <label className="p-3">Username</label>
          <input
            value={userName}
            onChange={handleUserNameChange}
            className="w-100 input-login"
            placeholder="Username"
          />
        </div>
        <div className="d-flex flex-column justify-center align-items-center w-100">
          <label className="p-3 ">Password</label>
          <input
            value={password}
            onChange={handlePasswordChange}
            className="w-100 input-login"
            placeholder="Password"
          />
        </div>
        <div className="login-button " onClick={(e) => login(e)}>
          Submit
        </div>
      </div>
    </div>
  );
};

export default connect()(Login);

// export default Login;
