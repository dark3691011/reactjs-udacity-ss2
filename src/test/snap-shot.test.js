import * as React from "react";
import { render } from "@testing-library/react";
import Login from "../components/Login";
import middleware from "../middleware";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers";
import { BrowserRouter as Router } from "react-router-dom";

const store = createStore(reducer, middleware);

describe("Greetings", () => {
  it("matches the snapshot when a name is passed", () => {
    var component = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
