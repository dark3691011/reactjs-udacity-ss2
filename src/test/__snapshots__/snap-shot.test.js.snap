// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greetings matches the snapshot when a name is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column justify-center align-items-center "
      >
        <h1>
          Employee Polls
        </h1>
        <img
          src="login_img.jpg"
        />
        <div
          class="d-flex flex-column justify-center align-items-center w-100"
        >
          <h3>
            Login
          </h3>
          
           
          <div
            class="d-flex flex-column justify-center align-items-center w-100"
          >
            <label
              class="p-3"
            >
              Username
            </label>
            <input
              class="w-100 input-login"
              placeholder="Username"
              value=""
            />
          </div>
          <div
            class="d-flex flex-column justify-center align-items-center w-100"
          >
            <label
              class="p-3 "
            >
              Password
            </label>
            <input
              class="w-100 input-login"
              placeholder="Password"
              value=""
            />
          </div>
          <div
            class="login-button "
          >
            Submit
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column justify-center align-items-center "
    >
      <h1>
        Employee Polls
      </h1>
      <img
        src="login_img.jpg"
      />
      <div
        class="d-flex flex-column justify-center align-items-center w-100"
      >
        <h3>
          Login
        </h3>
        
         
        <div
          class="d-flex flex-column justify-center align-items-center w-100"
        >
          <label
            class="p-3"
          >
            Username
          </label>
          <input
            class="w-100 input-login"
            placeholder="Username"
            value=""
          />
        </div>
        <div
          class="d-flex flex-column justify-center align-items-center w-100"
        >
          <label
            class="p-3 "
          >
            Password
          </label>
          <input
            class="w-100 input-login"
            placeholder="Password"
            value=""
          />
        </div>
        <div
          class="login-button "
        >
          Submit
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
